environments/commands/readiness_wait_command.py

```
@@ -0,0 +1,55 @@
+) -> int:
+    profile = load_profile(profile_name)
+    dependencies = resolve_dependencies(profile["dependencies"])
+    dependency_map = cast(Mapping[str, Mapping[str, object]], dependencies)
+    observations: dict[str, dict[str, object]] = {}
+    ready = True
+
+    for name in service_names:
+        if name not in INFERENCE_SERVICE_NAMES:
+            raise ProfileError(f"unsupported inference readiness service: {name}")
+        if name not in dependency_map:
+            raise ProfileError(f"{name} dependency is required")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)